### PR TITLE
feat(ui5-tree-item): add indeterminate state

### PR DESCRIPTION
**Pull request by nnaydenow**
Closes: #3618

### CODE REVIEW

File: packages/main/src/TreeItem.js [hunk 1]
```diff
@@ -29,6 +29,27 @@ const metadata = {
 		expanded: {
 			type: Boolean,
 		},
+		/**
+		* Defines whether the selection of tree node should be display as partially selected.
```
Comment (kineticjs) [inline packages/main/src/TreeItem.js:33]: minor: "should be displayed"
I guess a UA will review the texts in the end anyway?

Comment (xtatica) [inline packages/main/src/TreeItem.js:33]:
```suggestion
		* Defines whether the selection of a tree node is displayed as partially selected.
```

File: packages/main/src/TreeItem.js [hunk 2]
```diff
@@ -29,6 +29,27 @@ const metadata = {
...
+		* <br><br>
+		* <b>Note:</b> The indeterminate state can be set only programatically and can’t be achieved by user
+		* interaction and the resulting visual state depends on the values of the <code>indeterminate</code>
```
Comment (xtatica) [inline packages/main/src/TreeItem.js:36]:
```suggestion
		* interaction, meaning that the resulting visual state depends on the values of the <code>indeterminate</code>
```

File: packages/main/src/TreeItem.js [hunk 3]
```diff
@@ -29,6 +29,27 @@ const metadata = {
...
+		* and <code>selected</code> properties:
+		* <ul>
+		* <li> If the tree node is selected and indeterminate, it will be displayed as partially selected
```
Comment (xtatica) [inline packages/main/src/TreeItem.js:39]:
```suggestion
		* <li> If a tree node has both <code>selected</code> and <code>indeterminate</code> set to <code>true</code>, it is displayed as partially selected.
```

File: packages/main/src/TreeItem.js [hunk 4]
```diff
@@ -29,6 +29,27 @@ const metadata = {
...
+		* <li> If the tree node is selected and it is not indeterminate, it will be displayed as selected
```
Comment (xtatica) [inline packages/main/src/TreeItem.js:40]:
```suggestion
		* <li> If a tree node has <code>selected</code> set to <code>true</code> and <code>indeterminate</code> set to <code>false</code>, it is displayed as selected.
```

File: packages/main/src/TreeItem.js [hunk 5]
```diff
@@ -29,6 +29,27 @@ const metadata = {
...
+		* <li> If the tree node is not selected, it will be displayed as not selected regardless value of the indeterminate attribute
```
Comment (xtatica) [inline packages/main/src/TreeItem.js:41]:
```suggestion
		* <li> If a tree node has <code>selected</code> set to <code>false</code>, it is displayed as not selected regardless of the value of the <code>indeterminate</code> property.
```

File: packages/main/src/TreeItem.js [hunk 6]
```diff
@@ -29,6 +29,27 @@ const metadata = {
...
+		* </ul>
+		* <br>
+		* <b>Note:</b> This property will take affect only when the <code>ui5-tree</code> is in <code>MultiSelect</code> mode.
```
Comment (xtatica) [inline packages/main/src/TreeItem.js:44]:
```suggestion
		* <b>Note:</b> This property takes effect only when the <code>ui5-tree</code> is in <code>MultiSelect</code> mode.
```

**Review by xtatica — CHANGES_REQUESTED** [review]

**Review by kineticjs — APPROVED** [review]

**Review by xtatica — APPROVED** [review]

**Review by kineticjs — APPROVED** [review]

**Comment by nnaydenow** [comment]
Hi @ilhan007,

Can you take a look before merging it?

**Pull request merged**